_drafts/PR-blogs/planning-docs/example-code/animated-dots/original/script.js

```diff
@@ -0,0 +1,170 @@
+window.onload = init();
+function init() {
+  canvas = document.getElementById("canvas");
+  context = canvas.getContext("2d");
+  canvas.height = window.innerHeight;
+  canvas.width = window.innerWidth;
+  mouse = { x: 0, y: 0 };
+  colors = [
+    "#af0",
+    "#7CD14E",
+    "#1CDCFF",
+    "#FFFF00",
+    "#FF0000",
+    "#aee137",
+    "#bef202",
+    "#00b2ff",
+    "#7fff24",
+    "#13cd4b",
+    "#c0fa38",
+    "#f0a",
+    "#a0f",
+    "#0af",
+    "#000000",
+  ];
+  canvas.addEventListener("mousemove", MouseMove, false);
+  canvas.addEventListener("mousedown", MouseDown, false);
+  canvas.addEventListener("mouseup", MouseUp, false);
+  window.addEventListener("resize", canvasResize, false);
+  dotsHolder = [];
+  mouseMove = false;
+  mouseDown = false;
+  for (i = 0; i < 1000; i++) {
+    dotsHolder.push(new dots());
+  }
+
+  /*REQUEST ANIMATION FRAME*/
+  (function () {
+    var lastTime = 0;
+    var vendors = ["ms", "moz", "webkit", "o"];
+    for (var x = 0; x < vendors.length && !window.requestAnimationFrame; ++x) {
+      window.requestAnimationFrame =
+        window[vendors[x] + "RequestAnimationFrame"];
+      window.cancelAnimationFrame =
+        window[vendors[x] + "CancelAnimationFrame"] ||
+        window[vendors[x] + "CancelRequestAnimationFrame"];
+    }
+
+    if (!window.requestAnimationFrame)
+      window.requestAnimationFrame = function (callback, element) {
+        var currTime = new Date().getTime();
+        var timeToCall = Math.max(0, 16 - (currTime - lastTime));
+        var id = window.setTimeout(function () {
+          callback(currTime + timeToCall);
+        }, timeToCall);
+        lastTime = currTime + timeToCall;
+        return id;
+      };
+
+    if (!window.cancelAnimationFrame)
+      window.cancelAnimationFrame = function (id) {
+        clearTimeout(id);
+      };
+  })();
+}
+function canvasResize(event) {
+  canvas.height = window.innerHeight;
+  canvas.width = window.innerWidth;
+  cancelAnimationFrame(animate);
+}
+function MouseUp(event) {
+  if (mouseMove) {
+    mouseMove = false;
+  }
+  if (mouseDown) {
+    mouseDown = false;
+  }
+}
+function MouseDown(event) {
+  mouseDown = true;
+}
+function MouseMove(event) {
+  mouse.x = event.pageX - canvas.offsetLeft;
+  mouse.y = event.pageY - canvas.offsetTop;
+  if (mouseMove) {
+    context.lineTo(mouseX, mouseY);
+    context.stroke();
+  }
+}
+function dots() {
+  this.xPos = Math.random() * canvas.width;
+  this.yPos = Math.random() * canvas.height;
+  this.color = colors[Math.floor(Math.random() * colors.length)];
+  this.radius = Math.random() * 10;
+  this.vx = Math.cos(this.radius);
+  this.vy = Math.sin(this.radius);
+  this.stepSize = Math.random() / 10;
+  this.step = 0;
+  this.friction = 7;
+  this.speedX = this.vx;
+  this.speedY = this.vy;
+}
```

Comment:
I'd suggest writing this as a `makeDot` function which returns a simple object when called rather than defining your dots as classes: 

- As it is, you're using this class to fill the `dotsHolder` array by creating new instances of the dot class within a for loop, which is a functional programming-flavored technique as it is. 
- The class doesn't contain any state (all of the mutable state in the app, most notably `dotsHolder`, is defined with global variables already), so you don't get the benefit of treating a class as a stateful object either. 
- And because it's written as a class, you're encouraged to write things like `dots.draw` as prototype functions on the class itself. This couples `dots.draw` to the `dots` class, and can make it difficult in the future to decouple them if needed. Pure functions which aren't tied to a particular class are more portable and easier to adapt to new scenarios  as they arise. 

So! I'd suggest ditching the class and going with something like this instead: 

```js
const makeDot = () => {
  const radius = Math.random() * 10;

  return {
    xPos: Math.random() * canvas.width,
    yPos: Math.random() * canvas.height,
    radius: radius,
    vx: Math.cos(radius),
    vy: Math.sin(radius),
    // etc
  };
}

const drawDots = dot => {
  // dot drawing code
}
```